macros/edr/tests/on_run_end/handle_tests_results.sql

```diff
@@ -335,22 +338,18 @@
         'test_params': elementary.insensitive_get_dict_value(test_node, 'test_params'),
         'severity': elementary.insensitive_get_dict_value(test_node, 'severity'),
         'status': elementary.insensitive_get_dict_value(run_result_dict, 'status'),
-        'failures': elementary.insensitive_get_dict_value(run_result_dict, 'failures')
+        'failures': elementary.insensitive_get_dict_value(run_result_dict, 'failures'),
+        'samples': elementary.get_test_result_rows_as_dicts(test_node)
     }%}
     {{ return(test_result_dict) }}
 {% endmacro %}
 
-{% macro get_test_result_rows_as_dicts(flatten_test_node, sample_limit = none) %}
-    {% set test_row_dicts = [] %}
-    {% set test_compiled_code = elementary.get_compiled_code(flatten_test_node) %}
-    {% if test_compiled_code %}
-        {% if sample_limit %}
-            {% set test_compiled_code = test_compiled_code ~ ' limit ' ~ sample_limit %}
-        {% endif %}
-        {% set test_table_agate = run_query(test_compiled_code) %}
-        {% set test_row_dicts = elementary.agate_to_dicts(test_table_agate) %}
+{% macro get_test_result_rows_as_dicts(flatten_test_node) %}
+    {% set test_result = graph["elementary"]["test_results"].get(flatten_test_node.unique_id) %}
+    {% if not test_result %}
+      {{ return(none) }}
```

Comment:
maybe return empty list here so the code that parses it could just read a json on an empty list?

Reply:
I thought it would make more sense that a failed test would have `null` rather than an empty list which is like a passed test.